devcontainer/features/features.go

```diff
@@ -169,8 +169,12 @@ type Spec struct {
 
 // Extract unpacks the feature from the image and returns a set of lines
 // that should be appended to a Dockerfile to install the feature.
-func (s *Spec) Compile(options map[string]any) (string, error) {
-	var runDirective []string
+func (s *Spec) Compile(remoteUser string, options map[string]any) (string, error) {
+	// TODO figure out containerUser so we can set _CONTAINER_USER
```

Comment:
`containerUser` shouldn't be hard to determine - see [this code](https://github.com/coder/envbuilder/blob/cbbf185544e0104cce4b9c79078cebcdf33266d6/devcontainer/devcontainer.go#L151-L166).

The spec says it can also come from the `containerUser` property in `devcontainer.json`, which we aren't currently parsing out, so we should add that as well:

> The container user can be set with `containerUser` in the `devcontainer.json` and image metadata, `user` in the `docker-compose.yml`, `USER` in the Dockerfile, and can be passed down from the base image.

Reply:
Thanks, updated.  I think the logic we had was actually for `containerUser`.  The specs say remoteUser comes from the configs and containerUser comes from configs, USER instructions, or User in the image manifest.  If remoteUser is unset it defaults to containerUser.